From: Thomas Klute <thomas2.klute@uni-dortmund.de>
Date: Mon, 6 Feb 2017 17:11:47 +0100
Subject: Test suite: Log if a process to be stopped by PID file is not running

The runtests script calls "sleep ${TEST_QUERY_DELAY}" to keep the
input pipeline for gnutls-cli open, effectively creating a timeout of
TEST_QUERY_DELAY seconds for the HTTPS request. Normally the sleep
process is killed after the request completes to avoid stalling the
test suite. The sleep process no longer running at that point
indicates that the request timed out.
---
 test/runtests | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/runtests b/test/runtests
index 0020fb4..9e98b8d 100755
--- a/test/runtests
+++ b/test/runtests
@@ -88,6 +88,8 @@ function kill_by_pidfile()
 	local pid=$(cat "${pidfile}")
 	if [ -n "${pid}" ] && ps -p "${pid}"; then
 	    kill "${pid}"
+	else
+	    echo "No running process with PID ${pid} (${pidfile})."
 	fi
 	rm "${pidfile}"
     fi
